### PR TITLE
Consolidate `reward` calls into one

### PR DESCRIPTION
**Pull request by ureeves**
This is not only a performance optimization, but also useful for minimizing the amount of events emitted - which has a storage impact on prospective archivers.

### CODE REVIEW

**Review by herr-seppia — APPROVED** [review]
LGTM

File: contracts/stake/src/state.rs [hunk 1]
```diff
@@ -263,19 +265,23 @@ impl StakeState {
 
     /// Rewards a `account` with the given `value`. If a stake does not exist
```
Comment (Neotamandua) [inline contracts/stake/src/state.rs:266]:
```suggestion
    /// Rewards multiple accounts with the associated values based on the `rewards` being passed. If a stake does not exist
```

Now that account & value are no longer params, but encapsulated & consolidated in the rewards, we should make the comments a bit clearer. Added a suggestion for this

**Review by Neotamandua — APPROVED** [review]
LGTM. Added a small suggestion for the comment above reward()

File: contracts/stake/src/state.rs (hunk 1 above)
Comment (ureeves) [reply contracts/stake/src/state.rs:266]: @herr-seppia has commited to changing this in a future PR, which will implement event emission. I agree I could have done a better job, but for now I'm merging.

**Pull request merged**